Content.Server/DeviceNetwork/Components/DeviceNetworkComponent.cs

```diff
@@ -21,20 +20,74 @@ public enum ConnectionType
             Wireless,
             Apc
         }
+        // TODO allow devices to join more than one network?
+
+        // TODO if wireless/wired is determined by NetworkType, what is the point of WirelessNetworkComponent & the
+        // other network-type-specific components? Shouldn't DeviceNetId determine conectivity checks?
```

Comment:
"NetworkType" should be "ConnectionType" here. Will fix whenever I next commit other changes.

But yeahit bothers me that there are pre-defined networks that exist to separate devices, while there also exist completely independent network-type-components that exist to restrict connectivity.

IMO there shouldn't be a single "APC" network, APC device networks should get created dynamically whenever any entity with an ApcNetworkComponent connects to an `ApcNet` (the node-group / power network). That would also remove the need for the connectivity checks whenever messages get sent, because you would already know beforehand that any device on an apc network can connect to other devices on that network. I might try changing how that works & allowing devices to join more than one network at some point in a future PR.

Reply:
The deviceNetId is not a "connection type", it's used for segregating networks to prevent them from ever interacting with each other. Frequency might be changed by players but deviceNetId should not be shown to or changed by them. The enum is there to provide documented default deviceNetIds.

DeviceNetId should not be used to restrict connectivity. If you just dynamically create deviceNetIds you can't check for range or if a cable was cut etc. You also loose the ability to customize and combine device net components, which are not just "network-type" components. 

Currently there are only "connection type" components implemented but my plan is to have more types of device net components and to allow the corresponding systems to see and modify the payload.
The payload is missing from the BeforePacketSentEvent because I haven't added that yet.

One idea for a component I plan on adding is one that allows you to simulate encryption by wrapping parts of the payload in a class and adding a field to the payload containing a random string. As only primitive types are supposed to be shown to and modified by players they would see the random string instead of the original content them unless they have a special version of whatever will be used to sniff packets that also has that component.

Or a component that checks if both devices are on the same station.
The WiredNetworkComponent does that currently but only because the code checking if the sending and receiving devices are connected to the same PowerNet was broken and I had removed it for now.

You could then combine the WirelessNetworkComponent with the one that checks if you are on the same station to make a device that has a receiving range and only receives from devices on the same station.